frontend/src/modules/Metrics/Components/Metrics.tsx

```diff
@@ -0,0 +1,293 @@
+import { useState } from 'react'
+import MenuItem from '@mui/material/MenuItem'
+import { ReactComponent as LogoImg } from '@assets/img/lscicon.svg'
+import { useCourseValue } from '@context/CourseContext'
+import { useStudentValue } from '@context/StudentContext'
+import { AccountMenu } from 'Dashboard/Components/AccountMenu'
+import { Box, IconButton, SelectChangeEvent } from '@mui/material'
+import { StatGrid } from './StatGrid'
+import { MetricsTable } from './MetricsTable'
+import { DropdownSelect } from '@core/Components'
+import { DASHBOARD_PATH } from '@core/Constants'
+import { Link } from 'react-router-dom'
+
+export const Metrics = () => {
+  const { courses } = useCourseValue()
+  const { students } = useStudentValue()
+
+  function isDateInThisWeek(date: Date) {
+    const todayObj = new Date()
+    const todayDate = todayObj.getDate()
+    const todayDay = todayObj.getDay()
+    // get first date of week
+    const firstDayOfWeek = new Date(todayObj.setDate(todayDate - todayDay))
+    // get last date of week
+    const lastDayOfWeek = new Date(firstDayOfWeek)
+    lastDayOfWeek.setDate(lastDayOfWeek.getDate() + 6)
+    // if date is equal or within the first and last dates of the week
+    return date >= firstDayOfWeek && date <= lastDayOfWeek
+  }
+
+  const collegeAbbreviations: { [key: string]: string } = {
+    CALS: 'College of Agriculture and Life Sciences',
+    AAP: 'College of Architecture, Art, and Planning',
+    'A&S': 'College of Arts and Sciences',
+    Engineering: 'College of Engineering',
+    HumEc: 'College of Human Ecology',
+    Dyson:
+      'Dyson School of Applied Economics and Management (SC Johnson College of Business)',
+    Nolan:
+      'Nolan School of Hotel Administration (SC Johnson College of Business)',
+    ILR: 'School of Industrial and Labor Relations',
+    SCE: 'School of Continuing Education & Summer Sessions',
+    'Grad School': 'Graduate/Professional School',
+  }
+  const num_students_in_week = students.map((student) =>
+    student.groups.filter((membership) =>
+      isDateInThisWeek(membership.submissionTime)
+    )
+  )
+  //calculate number of unique students who have made requests
+  const num_students = {
+    number: students.length,
+    title: 'UNIQUE STUDENTS',
+    subtitle: 'made requests',
+    thisWeek: num_students_in_week.length,
+    showAdded: true,
+  }
+  const courses_in_week: string[] = []
+  num_students_in_week.map((member) =>
+    member.map((memberships) => {
+      if (courses_in_week.indexOf(memberships.courseId) === -1) {
+        courses_in_week.push(memberships.courseId)
+      }
+    })
```

Comment:
forgot to mention that you need to fix errors in the console. You don't have anything returning here I think so it's giving an error.